aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -411,11 +418,65 @@ Tensor & index_copy_(Tensor & self, int64_t dim, const Tensor & index, const Ten
   return at::_index_copy_(self, dim, index, source);
 }
 
+Tensor & _index_copy_impl_(Tensor & self, int64_t dim, const Tensor & index, const Tensor & source) {
+  // Handle the case when `self` is 0-dim
+  Tensor self_nonzero = self.dim() == 0 ? self.unsqueeze(-1) : self;
+
+  // The only different between the following  tensor iterator and that of index_fill_ is that
+  // this one has also source as an input. Maybe refactor?
```

Comment:
That is actually a good question. The issue is that it turns out `TensorIteratorConfig` is not copyable nor copy-assignable. So, it looks like only the restriding part could be shared. It is possible to write the same kernel, however, with the same kernel callback that returns a pointer to the output, source and index.